### PR TITLE
Support for windows endpoints

### PR DESCRIPTION
**Pull request by WardLT**
Adds support for Windows endpoints. 

The issues observed previously on Windows were the result of using Windows/DOS-style paths in the Globus SDK. This PR introduces a function that converts Windows to POSIX paths before performing a transfer.

### CODE REVIEW

File: mdf_toolbox/utils.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    Args:
+        path (str): Input path
+    Returns:
+        (str) Rectified path
+    """
+
+    is_windows = re.match('[A-Z]:\\\\', path) is not None
+    if is_windows:
+        ppath = PureWindowsPath(path)
+        return f'/{ppath.drive[:1].lower()}{ppath.as_posix()[2:]}'
```
Comment (jgaff) [inline mdf_toolbox/utils.py:26]: Python's f-strings were introduced in 3.6, but the rest of Toolbox works on previous versions. Can this be changed to `.format()` instead, to maintain compatibility?

Comment (WardLT) [reply mdf_toolbox/utils.py:26]: Sure, I can swap it over to ".format"

**Comment by WardLT** [comment]
We could have `rectify_path` in toolbox, but I'd prefer to to further complicate the toolbox.py with functions that are not intended to be used outside of `mdf_toolbox`. I'll defer to your answer, but feel that we could generally benefit from breaking the toolbox.py module into smaller, easier-to-digest chunks.

**Comment by jgaff** [comment]
Breaking up the Toolbox functions into separate files for organization purposes is something I've thought about. However, Toolbox is used pretty extensively throughout MDF projects (and some non-MDF projects), so I don't want to change how they're called - `mdf_toolbox.[utility]` is the pattern I'd like to stick with. So for now, let's put `rectify_path` in `toolbox.py` and I'll sort out the source code organization later.

As far as `rectify_path` not being intended for use outside of Toolbox, these sorts of issues have a way of cropping up repeatedly. While I don't have another use case for it right now, I would guess that it will be useful to another project in the future.

**Comment by WardLT** [comment]
Sure, I have moved it into toolbox.py.

A thought about organization, it could be useful to break the groups of utilities out into their own separate "*.py" files and then import from those directories from `mdf_toolbox/__init__.py`. That way, you could both maintain the simple `mdf_toolbox.function` syntax and also be able to search functions by their type (e.g., `mdf_toolbox.auth.login` for the login function).

**Comment by jgaff** [comment]
Thanks.
Yeah, that's the plan. The functions are already (roughly) grouped, I just have to get around to actually separating them out.

Oh, one last thing: Do you mind switching the PR to merge into the `toolbox-dev` branch? `master` is protected against merges without successful tests, but Travis tests don't run on external PRs (to protect private keys). I'll merge it into `master` from there and get it released.

**Comment by WardLT** [comment]
Done. I've changed it to go to toolbox-dev

**Pull request merged**